asf_public_discourse_home_decarbonisation/pipeline/bert_topic_analysis/bert_params_config.py

```diff
@@ -0,0 +1,110 @@
+"""
+File with parameters to test BERTopic to identify topics:
+- data_source_parms: data source related parameters
+- model_and_additional_params: model specificities and additional data filtering parameters
+"""
+
+data_source_parms = [
+    {
+        "data_source": "mse",
+        "slice": [
+            {
+                "category": "all",
+                "keywords": "heat_pump_keywords",
+            },
+            {
+                "category": "all",
+                "keywords": "boiler_keywords",
+            },
+            {
+                "category": "green-ethical-moneysaving",
+            },
+            {
+                "category": "lpg-heating-oil-solid-other-fuels",
+            },
+            {
+                "category": "is-this-quote-fair",
+            },
+            {
+                "category": "energy",
+            },
+            {
+                "category": "all",
+            },
+        ],
+    },
+    {
+        "data_source": "bh",
+        "slice": [
+            {
+                "category": "all",
+                "keywords": "heat_pump_keywords",
+            },
+            {
+                "category": "all",
+                "keywords": "boiler_keywords",
+            },
+            {
+                "category": "all",
+            },
+            {
+                "category": "119_air_source_heat_pumps_ashp",
+            },
+            # {
+            #     "category": "120_ground_source_heat_pumps_gshp",  # too small to generate topics
+            # },
+            # {
+            #     "category": "125_general_alternative_energy_issues",  # too small to generate topics
+            # },
+            # {
+            #     "category": "136_underfloor_heating",
+            # },
+            # {
+            #     "category": "137_central_heating_radiators",
+            # },
+            # {
+            #     "category": "139_boilers_hot_water_tanks",
+            # },
+            # {"category": "140_other_heating_systems"},
+        ],
+    },
+]
+
+from bertopic.representation import KeyBERTInspired
+from sklearn.feature_extraction.text import CountVectorizer
+
+# Additional parameters like the text column and filter are in the following dictionary instead of in the
+# data_source_parms because a different text column/filter would change the model specifities
```

Comment:
I think this is the fine for the moment and we can try to think of another way of doing it later! I guess you could have an additional dictionaries for 'text_column' and 'filter' to avoid repetitiveness when we start looking at the text and original_posts?

Reply:
I thought about that, but that also poses other problems, hence why I decided to keep it like this. Let's do as you say and keep it as it is for now